packages/govuk-frontend-review/src/views/examples/javascript-errors/index.njk

```diff
@@ -0,0 +1,55 @@
+{% extends "layouts/layout.njk" %}
+
+{% from "govuk/components/back-link/macro.njk" import govukBackLink %}
+{% from "govuk/components/character-count/macro.njk" import govukCharacterCount %}
+
+{% block beforeContent %}
+  {{ govukBackLink({
+    href: "/"
+  }) }}
+{% endblock %}
+
+{% block content %}
+  <h1 class="govuk-heading-l">JavaScript errors</h1>
+  <p class="govuk-body-lead">
+    Open your <a class="govuk-link" href="https://developer.mozilla.org/en-US/docs/Web/API/console#see_also">brower's console</a>
+    to witness the error thrown in browsers where GOV.UK Frontend is not supported.
+  </p>
+  {{ govukCharacterCount({
+    label: {
+      text: "Any other comment"
+    },
+    id: "some-content",
+    name: "some-content"
+  })}}
+{% endblock %}
+
+{% block bodyEnd %}
+  <script type="module" src="/javascripts/govuk-frontend.min.js"></script>
+
+  <!-- SupportError -->
+  <script type="module">
+    // Calling `initAll` would result in errors being logged rather than thrown
+    import { CharacterCount } from '/javascripts/govuk-frontend.min.js'
+
+    // Simulate GOV.UK Frontend not being supported
+    document.body.classList.remove('govuk-frontend-supported')
+
+    const $element = document.querySelector('[data-module="govuk-character-count"]')
+
+    try {
+      // Instantiate the component directly so errors are thrown
+      new CharacterCount($element)
+      // Use `finally` to ensure we tidy up regardless of the error
+    } finally {
```

Comment:
Do we have try/finally support in our supported browsers?

Or even try-without-catch for future browser testing?

Reply:
I'm not aware of any support issue with `try` blocks when running synchronous code. It's a feature that's been there [since pretty much the start](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch). [Using it without `catch` was there as early as ES5 in the specs as well](https://262.ecma-international.org/5.1/#sec-12.14) (maybe even earlier, but I did not find specs for ES3 quickly)

It's a different matter for asynchronous code, though, as [`Promise.finally` landed more recently](https://caniuse.com/?search=finally), so that's where we'll need to be a bit more cautious. Regarding `catch`, the more recent development is the ability to omit grabbing the `(error)`, for when you don't want to do anything with it (the ["Optional catch binding" in the MDN support table](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#browser_compatibility)), is that what you refer to by "try-without-catch"?

Reply:
>…is that what you refer to by "try-without-catch"?

Yeah that's the one https://caniuse.com/mdn-javascript_statements_try_catch_optional_catch_binding

Got a bit of a gap from [`<script type="module">`](https://caniuse.com/es6-module) to [`try...catch`: Optional catch binding](https://caniuse.com/mdn-javascript_statements_try_catch_optional_catch_binding)

Best adding an empty catch and running the tidy up below

Reply:
Lack of a `catch` clause has been well supported since ES5, so we're good here. The optional catch binding is about omitting the "argument" after `catch`. Here's a screenshot of the page in Chrome 61 (that doesn't support Optional catch binding):
- the error is logged in the console (showing the component was instantiated with the `govuk-frontend-supported` class removed)
- the `govuk-frontend-supported` class is on the `<body>` (after having been added back in the `finally` block)

<img width="1703" alt="" src="https://github.com/alphagov/govuk-frontend/assets/396367/f6f5b4a9-9506-4f38-8a1a-e404ccd105e0">

Reply:
That's good enough for me. Thanks for testing it @romaricpascal